test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -382,16 +382,78 @@ var _ = ginkgo.Describe("end to end testing", func() {
 			util.ValidateNumDeployments(ctx, wec1, ns, 1)
 			util.ValidateNumDeployments(ctx, wec2, ns, 0)
 			util.ValidateSingletonStatus(ctx, wds, ns, "nginx-singleton")
-			patch_again := []byte(`{"spec":{"clusterSelectors":[{"matchLabels":{"name":"CelestialNexus"}}]}}`)
+			patchAgain := []byte(`{"spec":{"clusterSelectors":[{"matchLabels":{"name":"CelestialNexus"}}]}}`)
 			_, err = ksWds.ControlV1alpha1().BindingPolicies().Patch(
-				ctx, "nginx-singleton", types.MergePatchType, patch_again, metav1.PatchOptions{})
+				ctx, "nginx-singleton", types.MergePatchType, patchAgain, metav1.PatchOptions{})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 			util.ValidateNumDeployments(ctx, wec1, ns, 0)
 			util.ValidateNumDeployments(ctx, wec2, ns, 0)
 			util.ValidateSingletonStatusZeroValue(ctx, wds, ns, "nginx-singleton")
 		})
 	})
 
+	ginkgo.Context("singleton status eventual consistency", func() {
+		ginkgo.It("cleans up previously synced but currently invalid singleton status", func(ctx context.Context) {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.CreateDeployment(ctx, wds, ns, "nginx-singleton",
+				map[string]string{
+					"app.kubernetes.io/name": "nginx-singleton",
+				})
+			util.CreateBindingPolicy(ctx, ksWds, "nginx-singleton",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"name": "cluster1"}},
+				},
+				[]ksapi.DownsyncPolicyClause{
+					{DownsyncObjectTest: ksapi.DownsyncObjectTest{
+						ObjectSelectors: []metav1.LabelSelector{{MatchLabels: map[string]string{"app.kubernetes.io/name": "nginx-singleton"}}},
+					}},
+				},
+			)
+			BPPatch := []byte(`{"spec":{"wantSingletonReportedState": true}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "nginx-singleton", types.MergePatchType, BPPatch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+			util.ValidateSingletonStatus(ctx, wds, ns, "nginx-singleton")
+			ginkgo.By("Singleton status synced")
+
+			// Restart the controller manager without starting the status controller.
+			util.ScaleDeployment(ctx, coreCluster, "wds1-system", "kubestellar-controller-manager", 0)
```

Comment:
Why scale down and up? Isn't the patch operation enough?

Reply:
If we use patch, then there will be an interval during which the terminating and old pods and the starting and new pods coexist. Sometimes the terminating pods last long enough, such that the old pods are still functioning during the simulated absence --- which makes the test flaky.

So I did scale down and up to ensure the status controller is indeed not functioning during the simulated absence.

Reply:
I just realized that I previously read that wrong. I thought the concern was that startup of the partially-disabled controller manager while the fully enabled one was also running would make the test flaky. I see now that the concern was that the test would proceed to probe behavior while both controller managers were running. I think that proper waiting in the transition code will rule out that scenario.